src/nvim/os/job.c

```diff
@@ -62,20 +83,22 @@ void job_init(void)
   uv_timer_init(uv_default_loop(), &job_stop_timer);
   uv_signal_init(uv_default_loop(), &schld);
   uv_signal_start(&schld, chld_handler, SIGCHLD);
+
+  job_map = pmap_new(int)();
 }
 
 /// Releases job control resources and terminates running jobs
 void job_teardown(void)
 {
   // Stop all jobs
-  for (int i = 0; i < MAX_RUNNING_JOBS; i++) {
-    Job *job;
-    if ((job = table[i]) != NULL) {
-      uv_kill(job->pid, SIGTERM);
-      job->term_sent = true;
-      job_stop(job);
-    }
-  }
+  Job *job;
+  int id; (void)id;
```

Comment:
Thanks for the tip about `map_foreach_value`!

> Wondering why there is (void)id.

I also ~~get~~ got this warning with debug mode.